src/components/GoogleMapsLink.js

```diff
@@ -20,15 +17,30 @@ class GoogleMapsLink extends React.Component {
     };
 
     googleMapsUrl(): string {
-        let toAddr = this.props.to;
-
-        let start = encodeURIComponent(this.props.from);
-        let query = encodeURIComponent(
-            `${toAddr.streetAddressLine1()} ${toAddr.streetAddressLine2()}`
+        const toAddr = this.props.to;
+        const {travelTime} = toAddr;
+        const mode = travelTime && travelTime.mode == "TRANSIT" ? "r" : "w";
+        const coords = storage.getJSON("coordinates");
+        const start = encodeURIComponent(
+            coords ? "Current Location" : `${storage.getItem("location")}`
+        );
```

Comment:
I'd always say "Current Location" for the link. Make it Google's problem.

Use case, I'm currently squatting in Footscray, but my Centrelink card says I live in the outer east, so that's where I need to go to get services. I type "Ringwood" into Izzy as my location, but I still need to get to the service from where I currently am.